.github/workflows/docker.yml

```diff
@@ -0,0 +1,71 @@
+
+name: Docker CI/CD
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '*'
+  pull_request:
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v2
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: ghcr.io/${{ github.repository }}
+
+      - name: Build and push Docker image
```

Comment:
@ksysoev I need to credentials setup to push to github registry. Could you help?
![image](https://github.com/user-attachments/assets/03191e43-f290-4fc6-a620-a33228a02f3f)

Reply:
@shan-96 gave write permissions for workflow, hope it should work.

Reply:
```
Error: buildx failed with: ERROR: failed to solve: failed to push ghcr.io/ksysoev/deriv-api-bff:pr-10: unexpected status from POST request to https://ghcr.io/v2/ksysoev/deriv-api-bff/blobs/uploads/: 403 Forbidden
```

this is the latest error. earlier it was 401

Reply:
@ksysoev I think I found the problem, I am trying to login using
```
name: Log in to GitHub Container Registry
uses: docker/login-action@v2
with:
    registry: ghcr.io
    username: ksysoev
    password: ${{ secrets.GITHUB_TOKEN }}
```

I think GITHUB_TOKEN is not set as a CI CD secret

Reply:
@shan-96 should not be the case, `secrets.GITHUB_TOKEN` should be there by default, if  workflow is configured 
<img width="780" alt="image" src="https://github.com/user-attachments/assets/8fb5384a-50a0-44b2-94bd-3c4bf2aa9718">

Reply:
@shan-96 based on documentation, you config file looks to correct, if it'll not work. lets just add build for now I'll try to debug push to registry separately.  Maybe I need to configure something else as well... I never used GitHub container registry before.

Reply:
@ksysoev another doubt, is this workflow permission same for all jobs?